### PR TITLE
[Sync V] Integrate Temporal Alignment for remaining non-behavior Interfaces

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
## Motivation

Follow-up to #237 specific to ophys interfaces

Also starts adding holistic per-interface, per-method unit tests instead of just relying on the demonstrative integration tests added in #237

Next follow-up will be other behavior interfaces

## Checklist

- [X] Have you thoroughly read our [Developer Guide](https://neuroconv.readthedocs.io/en/main/developer_guide.html)?
- [X] Have you ensured the PR description clearly describes the problem and solutions?
- [X] Have you checked to ensure that there aren't other open or previously closed [Pull Requests](https://github.com/catalystneuro/neuroconv/pulls) for the same change?
- [X] If this PR fixes an issue, is the first line of the PR description `fix #XXX` where `XXX` is the issue number?
- [X] Did you update the [CHANGLOG.md](https://github.com/catalystneuro/neuroconv/blob/main/CHANGELOG.md) file on your branch to describe your changes?

### CODE REVIEW

**Comment by bendichter** [comment]
Is there a way of accessing the original timestamps after alignment has taken place?

**Comment by CodyCBakerPhD** [comment]
> Is there a way of accessing the original timestamps after alignment has taken place?

Current approach is complete in-place replacement after alignment methods are called, echo'ing the timestamp vector operations from SpikeInterface and ROIExtractors

File: src/neuroconv/tools/testing/data_interface_mixins.py [hunk 1]
```diff
@@ -92,20 +94,69 @@ def check_extracted_metadata(self, metadata: dict):
...
-                self.run_conversion(nwbfile_path)
-                self.check_read_nwb(nwbfile_path=nwbfile_path)
+                self.nwbfile_path = str(self.save_directory / f"{self.data_interface_cls.__name__}_{num}.nwb")
+                self.run_conversion(nwbfile_path=self.nwbfile_path)
+                self.check_read_nwb(nwbfile_path=self.nwbfile_path)
+
+                # Temporal alignment checks
+                self.check_get_timestamps()
+                self.check_align_starting_time_internal()
+                self.check_align_starting_time_external()
```
Comment (CodyCBakerPhD) [inline src/neuroconv/tools/testing/data_interface_mixins.py:159]: This adds some essential checks for all interfaces using the new mixin

File: src/neuroconv/tools/testing/data_interface_mixins.py [hunk 2]
```diff
@@ -158,6 +228,28 @@ def check_read_nwb(self, nwbfile_path: str):
...
+                self.check_metadata()
+                self.nwbfile_path = str(self.save_directory / f"{self.data_interface_cls.__name__}_{num}.nwb")
+                self.run_conversion(nwbfile_path=self.nwbfile_path)
+                self.check_read_nwb(nwbfile_path=self.nwbfile_path)
+
+                # Temporal alignment checks
+                # Temporary override to disable failing multi-segment case
+                # self.check_get_timestamps()
+                # self.check_align_starting_time_internal()
+                # self.check_align_starting_time_external()
```
Comment (CodyCBakerPhD) [inline src/neuroconv/tools/testing/data_interface_mixins.py:251]: However, some of the previously added interfaces have certain problems, so adding temporary overrides on the child class mixin level (to be debugged and removed in subsequent PRs)

**Comment by CodyCBakerPhD** [comment]
@bendichter Ready for review

File: src/neuroconv/tools/testing/data_interface_mixins.py (hunk 1 above)
Comment (CodyCBakerPhD) [reply src/neuroconv/tools/testing/data_interface_mixins.py:159]: Everything except a handful of behavior interfaces not currently using this mixin, yes. See below for the temporary child-class overrides that disable it in failing cases

**Review by bendichter — APPROVED** [review]

**Pull request merged**